tests/agent_disconnect_test.go

```diff
@@ -74,10 +67,7 @@ func TestProxy_Agent_Disconnect_HTTP_Persistent_Connection(t *testing.T) {
 			close(stopCh)
 
 			// Wait for the agent to disconnect
-			wait.Poll(100*time.Millisecond, 5*time.Second, func() (bool, error) {
-				ready, _ := proxy.server.Readiness.Ready()
-				return !ready, nil
-			})
+			waitForBackends(t, 0, proxy.server)
```

Comment:
Can we pass in a contextual message for the fatal case to understand what was going in when waitForBackends was called?

Reply:
I used `t.Helper()` to log the calling line context instead.

With `t.Helper()`, the failures look like this:
```
--- FAIL: TestProxy_Agent_Disconnect_HTTP_Persistent_Connection (0.00s)
    --- FAIL: TestProxy_Agent_Disconnect_HTTP_Persistent_Connection/grpc (0.00s)
        proxy_test.go:642: got 0 backends
        agent_disconnect_test.go:53: Error waiting for backend count: timed out waiting for the condition
```

Without it they look like this:
```
--- FAIL: TestProxy_Agent_Disconnect_HTTP_Persistent_Connection (0.00s)
    --- FAIL: TestProxy_Agent_Disconnect_HTTP_Persistent_Connection/grpc (0.00s)
        proxy_test.go:642: got 0 backends
        proxy_test.go:646: Error waiting for backend count: timed out waiting for the condition
```